src/main.rs

```diff
@@ -127,7 +144,9 @@ fn work() -> Result<(), Cue2CCDError> {
 
     let disc = Disc::from_cuesheet(cd, root);
     for sector in disc.sectors() {
-        sub_write.write_all(&sector.generate_subchannel())?;
+        sub_write
+            .write_all(&sector.generate_subchannel(&chosen_protection_type))
+            .expect("This should never be modified");
```

Comment:
Was the move to `expect()` here intentional? Mind swapping it back?

For context, the question mark operator is used for error handling; it's the way that `Result` types ("this could be an error" types) are usually handled. It checks if the value is an error - if it is, it immediately returns the error (like raising an exception), and otherwise it unwraps the `Result` to the inner okay value.

`expect()` panics (eg hard-crashes the program) if the `Result` contained an error, rather than raising the error. It's used mainly in places where you expect it *never* to return an error, and so if something does go wrong it's a serious problem you want to crash the program.

Reply:
I did it because at some point the ? stopped working, but that was probably due to other issues in what I had programmed. ? works fine now, so I'll go put that back.